### PR TITLE
Cumulus 1896

### PR DESCRIPTION
**Pull request by Jkovarik**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ All notable changes to this project are documented in this file.
...
+
+## [v1.3.0] 2020-02-14
+
+### Added
+
+- **CUMULUS-1896**
+  - Added a 'streaming' mode to the CMA.   This will allow client libraries to avoid requiring three subprocess/python invocations when running the CMA commands in sequence, thus lowering
+    the overhead of utilizing the CMA.
+
+### BREAKING CHANGES
```
Comment (markdboyd) [inline CHANGELOG.md:16]: I would put the breaking changes at the top of the version notes

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,20 @@ All notable changes to this project are documented in this file.
...
+
+- **CUMULUS-1896**
+  - Updated CMA precompiled binary files to a layer subdirectory `./cma_bin`. Users who utilize the pre-compiled binary should update their pathing expectations to that directory.   Users utilizing the distribution zip file as a layer should update their CMA executable target to `/opt/cma` on AWS.
```
Comment (markdboyd) [inline CHANGELOG.md:19]: I don't grok what this means:

> Users who utilize the pre-compiled binary should update their pathing expectations to that directory

Is it the same thing as the next line about updating the CMA executable target?

Also, is there any other way we can better advertise these kinds of differences in the Cumulus core docs? As someone who recently emerged from a fun round of CMA debugging, I worry about a note like this getting buried in release docs. I wonder if the Cumulus docs could at least specify the different value of `CUMULUS_MESSAGE_ADAPTER_DIR` for different CMA releases

File: __main__.py [hunk 3]
```diff
@@ -5,36 +5,109 @@
...
+            if not command:
+                command = next_line.strip()
+                sys.stderr.write(f'warning setting command to {command}\n')
+            else:
+                buffer += next_line
+
+
+def singleCommand(functionName):
+    """Executes a single CMA command"""
+    allInput = json.loads(input())
```
Comment (markdboyd) [inline __main__.py:84]: So you can still run a single command using the old style of input where the JSON has to be all on one line?

File: tests/test-program.py [hunk 4]
```diff
@@ -51,7 +53,87 @@ def execute_command(self, cmd, input_message):  # pylint: disable=no-self-use
...
+        """
+        Given a stdin pipe for a subprocess, write command/proc input to CMA subprocess
+        """
+        p_stdin.write((command + "\n").encode('utf-8'))
+        p_stdin.write(json.dumps(proc_input).encode('utf-8'))
+        p_stdin.write('\n'.encode('utf-8'))
+        p_stdin.write('<EOC>\n'.encode('utf-8'))
+        p_stdin.flush()
+
+    def transform_messages_streaming(self, testcase, context=None):
```
Comment (markdboyd) [inline tests/test-program.py:85]: Very nice. So we're just testing that running the full set of CMA commands in the stream on an input message returns the expected output? And that the stream exits properly

File: CONTRACT.md [hunk 5]
```diff
@@ -37,11 +39,40 @@ python ./cumulus-message-adapter.zip createNextEvent
 
 These functions should be run in the order outlined above. The output of `loadAndUpdateRemoteEvent` should be sent as `<event_json>` to `createNextEvent`. The output of the `loadNestedEvent` should be fed to a "business function" and the output should be the `<handler_response_json>` sent to `createNextEvent`. More details on these values is provided in sections below.
 
+## Streaming Interface
+
+The CMA also offers a streaming interface that utilizes the commands listed above, but allows for a command/response to be issued without incurring the overhead of loading a subprocess/reloading python dependencies repeatedly:
+
+```bash
+python ./cumulus_message_adapter.zip streaming
```
Comment (markdboyd) [inline CONTRACT.md:47]:
```suggestion
python ./cumulus_message_adapter.zip stream
```

I think that is correct based on the code below?

File: __main__.py (hunk 3 above)
Comment (markdboyd) [reply __main__.py:84]: Indeed, I just tried this

File: CONTRACT.md (hunk 5 above)
Comment (markdboyd) [reply CONTRACT.md:47]: I just tested and this seems to be the case

File: CHANGELOG.md (hunk 2 above)
Comment (Jkovarik) [reply CHANGELOG.md:19]: Remove "Users utilizing the distribution zip file as a layer should update their CMA executable target to `/opt/cma` on AWS."

Comment (Jkovarik) [reply CHANGELOG.md:19]: re: your comment, yeah, I think we need to add a doc PR in addition to CHANGELOG commentary in Core for core users.

File: tests/test-program.py (hunk 4 above)
Comment (Jkovarik) [reply tests/test-program.py:85]: Yes, indeed.    We're running the same set of inputs against both stream and individual command flows.  They should have the exact same output (although one obviously runs faster)

File: CHANGELOG.md (hunk 1 above)
Comment (Jkovarik) [reply CHANGELOG.md]: f1272be

File: CHANGELOG.md (hunk 2 above)
Comment (Jkovarik) [reply CHANGELOG.md]: f1272be

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,9 +1,25 @@
...
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+
+## [v1.3.0] 2020-02-14
+
+### BREAKING CHANGES
+
+- **CUMULUS-1896**
+  - Updated CMA precompiled binary files to a layer subdirectory `./cma_bin`.   Clients who utilize the pre-compiled binary should update their pathing expectations to make use of that directory.
```
Comment (markdboyd) [inline CHANGELOG.md:15]:
```suggestion
  - CMA precompiled binary files are now copied into a subdirectory `./cma_bin`.  Clients who utilize the pre-compiled binary should update their pathing expectations to make use of that directory. If you are using the CMA via [`cumulus-message-adapter-js`](https://github.com/nasa/cumulus-message-adapter-js), then this change is handled for you transparently as of version 1.2.0.
```

**Pull request merged**